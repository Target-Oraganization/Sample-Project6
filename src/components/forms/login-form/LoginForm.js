// @flow
import React from 'react';
import { View } from 'react-native';

import FormTextInput from '../form-text-input/FormTextInput';
import FormField from '../form-field/FormField';
import Button from '../../button/Button';

type Props = {
  email: ?string,
  password: ?string,
  onShouldChangeEmail: (email: string) => void,
  onShouldChangePassword: (password: string) => void,
  onShouldSubmit: () => void,
};

const styles = {
  container: {
    flex: 1,
  },
  formField: {
    flex: 1,
    maxHeight: 75,
    marginVertical: 10,
  },
  submitButton: {
    flex: 1,
    maxHeight: 55,
    marginVertical: 25,
  },
};

export default function LoginForm(props: Props) {
  return (
    <View style={styles.container}>
      <FormField labelText="Email" style={styles.formField}>
        <FormTextInput
          autoCapitalize="none"
          autoCorrect={false}
          value={props.email}
          keyboardType="email-address"
          onShouldChangeValue={props.onShouldChangeEmail}
        />
      </FormField>
      <FormField labelText="Password" style={styles.formField}>
        <FormTextInput
          autoCapitalize="none"
          autoCorrect={false}
          secureTextEntry
          value={props.password}
          onShouldChangeValue={props.onShouldChangePassword}
        />
      </FormField>
      <Button
        style={styles.submitButton}
        text="Login"
        onPress={props.onShouldSubmit}
      />
    </View>
  );
}
