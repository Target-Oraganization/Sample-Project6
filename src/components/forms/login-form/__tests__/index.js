// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import noop from 'lodash/noop';

import LoginForm from '../LoginForm';

test('renders as expected', () => {
  const tree = renderer
    .create(
      <LoginForm
        email="jpbrennecke@gmail.com"
        password="password"
        isValidEmail
        isValidPassword
        onShouldChangeEmail={noop}
        onShouldChangePassword={noop}
        onShouldSubmit={noop}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
