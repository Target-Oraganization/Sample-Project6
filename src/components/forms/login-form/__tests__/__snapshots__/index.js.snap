// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "minHeight": 75,
        },
        Object {
          "flex": 1,
          "marginVertical": 10,
          "maxHeight": 75,
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "#3B3E6F",
          "fontFamily": "PT Sans",
          "fontSize": 11,
          "letterSpacing": 1.8,
          "marginBottom": 7,
        }
      }
    >
      EMAIL
    </Text>
    <View
      style={
        Object {
          "backgroundColor": "#B7BAE1",
          "borderColor": "transparent",
          "borderRadius": 7,
          "borderStyle": "solid",
          "borderWidth": 1,
          "flex": 1,
          "paddingHorizontal": 20,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        autoCorrect={false}
        autoFocus={true}
        keyboardType="email-address"
        onChangeText={[Function]}
        style={
          Array [
            Object {
              "color": "#3B3E6F",
              "flex": 1,
              "fontFamily": "PT Sans",
              "fontSize": 17,
            },
            undefined,
          ]
        }
        underlineColorAndroid="transparent"
        value="jpbrennecke@gmail.com"
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "minHeight": 75,
        },
        Object {
          "flex": 1,
          "marginVertical": 10,
          "maxHeight": 75,
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "#3B3E6F",
          "fontFamily": "PT Sans",
          "fontSize": 11,
          "letterSpacing": 1.8,
          "marginBottom": 7,
        }
      }
    >
      PASSWORD
    </Text>
    <View
      style={
        Object {
          "backgroundColor": "#B7BAE1",
          "borderColor": "transparent",
          "borderRadius": 7,
          "borderStyle": "solid",
          "borderWidth": 1,
          "flex": 1,
          "paddingHorizontal": 20,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        autoCorrect={false}
        onChangeText={[Function]}
        secureTextEntry={true}
        style={
          Array [
            Object {
              "color": "#3B3E6F",
              "flex": 1,
              "fontFamily": "PT Sans",
              "fontSize": 17,
            },
            undefined,
          ]
        }
        underlineColorAndroid="transparent"
        value="password"
      />
    </View>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#3B3E6F",
        "borderRadius": 7,
        "elevation": 1,
        "flex": 1,
        "justifyContent": "center",
        "marginVertical": 25,
        "maxHeight": 55,
        "minHeight": 34,
        "opacity": 1,
        "paddingHorizontal": 37,
        "paddingVertical": 7,
      }
    }
  >
    <Text
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#FFF",
            "fontFamily": "PT Sans",
            "fontSize": 15,
            "letterSpacing": 1.8,
          },
          undefined,
        ]
      }
    >
      LOGIN
    </Text>
  </View>
</View>
`;
